option.go

```diff
@@ -99,6 +102,27 @@ func BlockHook(blockHook BlockHookFunc) Option {
 	}
 }
 
+// SegmentBlockHook adds a hook that runs when a block is received along with a SegmentSyncActions.
+// Note that if segmented sync is disabled, calls on SegmentSyncActions will have no effect.
+// See: SegmentSyncActions, SegmentDepthLimit.
+func SegmentBlockHook(sbh SegmentBlockHookFunc) Option {
```

Comment:
The idea with this hook is to provide a way for the client to decode a synced CID and set the next segment sync CID from it. 

In general, I have taken a conservative approach of requiring explicit config over trying to auto-detect things since we don't know the shape of the general DAG being synced and there are a lot of ways things can go wrong.